### PR TITLE
3 sum closest program is wrong!

### PR DESCRIPTION
**Pull request by yqtaowhu**
hello ,my dear friend :
         in this cpp,you write
     if (sum - target> 0) {
					     					while (high>0 && num[high] == num[high - 1]) high--;
     					high--;
				}
				i think is wrong ,because the high should not higer then low,
for example if araay is [1,1,1-1], you answer is wrong
i think should change the high>low
the next loop while is also change as low <high

### CODE REVIEW

File: 3Num.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+				if (sum - target> 0) {
+					//skip the duplication
+					while (high>low && num[high] == num[high - 1]) high--;
+					//move the `high` pointer
+					high--;
+				}
+				else {
+					//skip the duplication
+					while (low<high && num[low] == num[low + 1]) low++;
```
Comment (yqtaowhu) [inline 3Num.cpp]: this is wrong

File: 3Num.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+				//tracking the minmal distance
+				if (abs(sum - target) < distance) {
+					distance = abs(sum - target);
+					result = sum;
+				}
+
+				if (sum - target> 0) {
+					//skip the duplication
+					while (high>low && num[high] == num[high - 1]) high--;
```
Comment (yqtaowhu) [inline 3Num.cpp]: this should change as high>low

**Comment by haoel** [comment]
It's fine, I have a `while (high > low)` outside! so, in the nested while-loop, it's unnecessary to check.. ;-)

**Comment by yqtaowhu** [comment]
thank you for your reply,
yes,you are right,i am think too much,sorry ..............................................................

**Comment by brainblowjob** [comment]
kek

**Pull request closed without merging**